src/CommandHandler/Wishlist/AddProductToSelectedWishlistHandler.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+/*
+ * This file was created by developers working at BitBag
+ * Do you need more information about us and what we do? Visit our https://bitbag.io website!
+ * We are hiring developers from all over the world. Join us and start your new, exciting adventure and become part of us: https://bitbag.io/career
+*/
+
+declare(strict_types=1);
+
+namespace BitBag\SyliusWishlistPlugin\CommandHandler\Wishlist;
+
+use Symfony\Component\HttpKernel\Exception\NotFoundHttpException;
+use BitBag\SyliusWishlistPlugin\Command\Wishlist\AddProductToSelectedWishlist;
+use BitBag\SyliusWishlistPlugin\Entity\WishlistProductInterface;
+use BitBag\SyliusWishlistPlugin\Factory\WishlistProductFactoryInterface;
+use Doctrine\ORM\EntityManagerInterface;
+use Symfony\Component\Messenger\Handler\MessageHandlerInterface;
+
+
+final class AddProductToSelectedWishlistHandler implements MessageHandlerInterface
+{
+    private WishlistProductFactoryInterface $wishlistProductFactory;
+
+    private EntityManagerInterface $wishlistManager;
+
+    public function __construct(
+        WishlistProductFactoryInterface $wishlistProductFactory,
+        EntityManagerInterface $wishlistManager
+    ) {
+        $this->wishlistProductFactory = $wishlistProductFactory;
+        $this->wishlistManager = $wishlistManager;
+    }
+
+    public function __invoke(AddProductToSelectedWishlist $addProductToSelectedWishlist)
+    {
+        $product = $addProductToSelectedWishlist->getProduct();
+        $wishlist = $addProductToSelectedWishlist->getWishlist();
+
+        if (null === $product) {
+            throw new NotFoundHttpException();
+        }
+
+        /** @var WishlistProductInterface $wishlistProduct */
+        $wishlistProduct = $this->wishlistProductFactory->createForWishlistAndProduct($wishlist, $product);
+
+        $wishlist->addWishlistProduct($wishlistProduct);
+        $this->wishlistManager->flush();
```

Comment:
Sylius Repository has `add` method which is better in terms of domain (you're not dealing with an implementation).

Reply:
Why its better?